addons/pos_restaurant/static/src/app/services/pos_store.js

```diff
@@ -527,6 +523,7 @@ patch(PosStore.prototype, {
     },
     async submitOrder() {
         const order = this.getOrder();
+        await this.ensureGuestCustomerCount(order);
```

Comment:
We should check the return of this method and stop the process if no guest nb selected